src/anomalib/trainer/utils/thresholder.py

```diff
@@ -86,17 +73,45 @@ def update(self, outputs: STEP_OUTPUT) -> None:
         Args:
             outputs (STEP_OUTPUT): Step outputs.
         """
-        if self.threshold_method == ThresholdMethod.ADAPTIVE:
-            self._update_thresholds(self.trainer.image_threshold, self.trainer.pixel_threshold, outputs)
-
-    @staticmethod
-    def _update_thresholds(
-        image_metric: AnomalyScoreThreshold,
-        pixel_metric: AnomalyScoreThreshold,
-        outputs: STEP_OUTPUT,
-    ) -> None:
+        image_metric = self.trainer.image_threshold
+        pixel_metric = self.trainer.pixel_threshold
         image_metric.cpu()
         image_metric.update(outputs["pred_scores"], outputs["label"].int())
-        if "mask" in outputs.keys() and "anomaly_maps" in outputs.keys():
+        if (
+            self.trainer.task_type != TaskType.CLASSIFICATION
+            and "anomaly_maps" in outputs.keys()
+            and "mask" in outputs.keys()
+        ):
             pixel_metric.cpu()
+            # TODO this should use bounding boxes for detection task type
             pixel_metric.update(outputs["anomaly_maps"], outputs["mask"].int())
+
+    def _get_threshold_method(self, threshold_method: dict | None) -> BaseAnomalyScoreThreshold:
```

Comment:
how about naming this `def _get_thresholder`? It is a bit confusing to have the same name as the arg? Thoughts?

Reply:
I agree that the current naming is confusing, but I feel that `_get_thresholder` is also confusing, because the class itself is also called `Thresholder`. Not sure what would be the best option.

Reply:
Maybe `_get_threshold_metric`. After all, the returned instance is a subclass of `Metric`